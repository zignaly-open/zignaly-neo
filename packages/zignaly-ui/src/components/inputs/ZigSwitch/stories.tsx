import { Meta, StoryObj } from "@storybook/react";
import ZigSwitch from "./index";

const meta = {
  title: "Inputs/ZigSwitch",
  component: ZigSwitch,
} as Meta;
export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
