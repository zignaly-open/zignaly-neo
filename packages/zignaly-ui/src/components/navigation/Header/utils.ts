// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
