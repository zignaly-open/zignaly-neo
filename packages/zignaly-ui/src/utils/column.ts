import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
