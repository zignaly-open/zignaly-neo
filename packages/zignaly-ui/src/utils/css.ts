import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
