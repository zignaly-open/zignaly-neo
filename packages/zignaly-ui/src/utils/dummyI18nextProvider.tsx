import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
