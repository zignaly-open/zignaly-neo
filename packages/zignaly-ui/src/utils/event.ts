export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
