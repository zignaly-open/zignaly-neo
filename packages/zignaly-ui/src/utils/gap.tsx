import styled from "styled-components";

export const Gap = styled.div<{ gap: number }>`
  ${(props) => `
  padding: ${props.gap}px;
  `}
`;
