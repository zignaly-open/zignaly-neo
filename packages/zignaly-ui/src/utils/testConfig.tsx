import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
