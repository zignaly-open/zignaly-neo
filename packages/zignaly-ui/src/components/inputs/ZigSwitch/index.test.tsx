import React from "react";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { fireEvent, screen, waitFor } from "@testing-library/react";
import ZigSwitch from ".";

describe("components/inputs/ZigSwitch", () => {
  it("should render without crashing", () => {
    const { container } = renderWithProvidersUi(<ZigSwitch />);
    const switchContainer = container.querySelector(".MuiSwitch-root") as HTMLElement;
    expect(switchContainer).toBeVisible();
    expect(switchContainer).toMatchSnapshot();
  });

  it("should handle switch state change", async () => {
    const onChangeMock = jest.fn();
    renderWithProvidersUi(<ZigSwitch onChange={onChangeMock} />);
    const switchInput = screen.getByRole("checkbox");
    fireEvent.click(switchInput);

    await waitFor(() => {
      expect(onChangeMock).toHaveBeenCalled();
    });
  });
});
