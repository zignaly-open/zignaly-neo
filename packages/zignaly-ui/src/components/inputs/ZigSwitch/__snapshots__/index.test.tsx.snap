// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ZigSwitch should render without crashing 1`] = `
<span
  class="MuiSwitch-root MuiSwitch-sizeMedium css-1xgfkn7-MuiSwitch-root"
>
  <span
    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1hphgtg-MuiButtonBase-root-MuiSwitch-switchBase"
  >
    <input
      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
      type="checkbox"
    />
    <span
      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    class="MuiSwitch-track css-dnkzfa-MuiSwitch-track"
  />
</span>
`;
